service/lib/agama/dbus/software/manager.rb

```diff
@@ -67,6 +67,24 @@ def issues
         private_constant :SOFTWARE_INTERFACE
 
         dbus_interface SOFTWARE_INTERFACE do
+          # array of repository properties: pkg-bindings ID, alias, name, URL, product dir, enabled
+          # and loaded flag
+          dbus_method :ListRepositories, "out Result:a(issssbb)" do
```

Comment:
In storage we are tending to other kind of interfaces. The storage API returns JSON data instead of D-Bus structs, dicts, etc. At the end, we are using D-Bus as a transport layer, and probably we will replace it in the future by a HTTP API. Having JSON data has 2 advandanges in our case:

* Replacing D-Bus by HTTP will be almost transparent.
* The rust code is only a proxy that defines an endpoint (it does not need to interpret the data coming from D-Bus).

I guess it is fine to continue with this approach using structs in software. Anyway, we want to rewrite this service in rust.